Support Onboarding & minor changes

ilagomatis opened the pull request:
- Add onboard endpoint (ota-service)
- Increase app size in partition table
- Fix DICE specific problems (ota-service)
- Fix MAC retrieval

ilagomatis committed:
feat: Support onboard endpoint

 - ota-service: Updated submodule version
 - esp32-akri: Update submodule version

PR: https://github.com/nubificus/esp32-ota-update/pull/9
Signed-off-by: Ilias Lagomatis <ilago@nubificus.co.uk>
Reviewed-by: Anastassios Nanos <ananos@nubificus.co.uk>
Approved-by: Anastassios Nanos <ananos@nubificus.co.uk>

ilagomatis committed:
Add custom partition table

I noticed that the generated firmware image
for esp32c6 did not fit on the default size
application partition.

The custom partition table supports at least
8MB flashes.

PR: https://github.com/nubificus/esp32-ota-update/pull/9
Signed-off-by: Ilias Lagomatis <ilago@nubificus.co.uk>
Reviewed-by: Anastassios Nanos <ananos@nubificus.co.uk>
Approved-by: Anastassios Nanos <ananos@nubificus.co.uk>

ilagomatis committed:
refactor: Update submodules

- Support onboarding
- Minor fixes: Dice Hashes & MAC retrieval

PR: https://github.com/nubificus/esp32-ota-update/pull/9
Signed-off-by: Ilias Lagomatis <ilago@nubificus.co.uk>
Reviewed-by: Anastassios Nanos <ananos@nubificus.co.uk>
Approved-by: Anastassios Nanos <ananos@nubificus.co.uk>

ananos reviewed (approved):
thanks @ilagomatis.

Would it make sense to have an sdkconfig.defaults per device? I think it is [supported](https://docs.espressif.com/projects/idf-build-apps/en/latest/explanations/config_rules.html). We could have for instance: 
`sdkconfig.defaults.esp32`: `CONFIG_ESPTOOLPY_FLASHSIZE_4MB=y`
`sdkconfig.defaults`: `CONFIG_ESPTOOLPY_FLASHSIZE_8MB=y`

etc.
If we have another way of customizing/automating this, no prob.

ilagomatis committed:
ci: Update runners

PR: https://github.com/nubificus/esp32-ota-update/pull/9
Signed-off-by: Ilias Lagomatis <ilago@nubificus.co.uk>
Reviewed-by: Anastassios Nanos <ananos@nubificus.co.uk>
Approved-by: Anastassios Nanos <ananos@nubificus.co.uk>

ilagomatis committed:
feat: Introduce per-device partition table

Currently, devices of type esp32 require a different
partition table than the rest ones, which have >=8MB
flash size. For that reason we introduce an esp32-spe-
cific sdkconfig defaults file.

PR: https://github.com/nubificus/esp32-ota-update/pull/9
Signed-off-by: Ilias Lagomatis <ilago@nubificus.co.uk>
Reviewed-by: Anastassios Nanos <ananos@nubificus.co.uk>
Approved-by: Anastassios Nanos <ananos@nubificus.co.uk>

ananos reviewed (approved):
thanks @ilagomatis!
